src/main/java/model/UserModel.java

```diff
@@ -28,17 +28,30 @@ public int checkParameters(Map<String, String[]> parameters) {
 	private int checkLoginData(String username, String password) {
 		int result = 0;
 		ResultSet data = daoFactory.getUserDao().getLoginData(username);
+
+		try {
+			data.next();
+			if(Password.checkPassword(password, data.getString(2), data.getString(3))) {
+				result = data.getInt(1);
+			}
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+		return result;
+	}
+	
+	public boolean checkUser(String username) {
+		boolean result = false;
+		ResultSet data = daoFactory.getUserDao().checkUser(username);
 		
-		if(data != null) {
-			try {
-				data.next();
-				if(Password.checkPassword(password, data.getString(2), data.getString(3))) {
-					result = data.getInt(1);
-				}
-			} catch (SQLException e) {
-				e.printStackTrace();
+		try {
```

Comment:
I think `data` should be tested for `null` value before the `try-catch` block to avoid an exception.

Reply:
same as in the above example, in java executeQuery method will never return a null so it is unnecessary. It should be checked with data.next()

Reply:
Not in case of the `SQLException` 😄